src/rooms/rooms.service.ts

```diff
@@ -112,10 +110,30 @@ export class RoomsService {
         message: 'Not found stack with this id or this stack is not active',
       });
     }
+    if (questionStack.status != QuestionStackStatus.ACTIVE) {
+      errs.push({
+        at: 'questionStackStatus',
+        message: 'Question stack is USED or not ACTIVE!',
+      });
+    }
+    if (info.type != questionStack.type) {
+      errs.push({
+        at: 'questionStackType',
+        message: "Question stack's and room's type are not match!",
+      });
+    }
+
     if (errs.length > 0) {
       return [null, errs];
     }
     try {
+      await this.questionStackRepository
+        .createQueryBuilder()
+        .update(questionStack)
+        .set({ status: QuestionStackStatus.USED })
+        .where('id= :id', { id: questionStack.id })
+        .execute();
+
```

Comment:
Put this will make this query can't roll back when bug come

Reply:
resolved